lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -0,0 +1,515 @@
+    , ClientStNext (..)
+    , chainSyncClientPeer
+    )
+import Ouroboros.Network.Protocol.ChainSync.Codec
+    ( codecChainSync )
+import Ouroboros.Network.Protocol.ChainSync.Type
+    ( ChainSync )
+import Ouroboros.Network.Protocol.Handshake.Version
+    ( DictVersion (..), simpleSingletonVersions )
+import Ouroboros.Network.Protocol.LocalTxSubmission.Client
+    ( LocalTxSubmissionClient (..), localTxSubmissionClientPeer )
+import Ouroboros.Network.Protocol.LocalTxSubmission.Codec
+    ( codecLocalTxSubmission )
+import Ouroboros.Network.Protocol.LocalTxSubmission.Type
+    ( LocalTxSubmission )
+
+import Control.Exception
+    ( catch, throwIO )
+import Control.Monad.Class.MonadST
+    ( MonadST )
+import Control.Monad.Class.MonadThrow
+    ( MonadThrow )
+import Control.Monad.Class.MonadTimer
+    ( MonadTimer )
+import Control.Tracer
+    ( Tracer, contramap )
+import Data.ByteString.Lazy
+    ( ByteString )
+import Data.Text
+    ( Text )
+import Data.Void
+    ( Void )
+import Network.Socket
+    ( AddrInfo (..), Family (..), SockAddr (..), SocketType (..) )
+
+import qualified Cardano.Chain.Block as CC
+import qualified Cardano.Chain.Common as CC
+import qualified Cardano.Chain.UTxO as CC
+import qualified Cardano.Crypto as CC
+import qualified Cardano.Wallet.Primitive.Types as W
+import qualified Codec.Serialise as CBOR
+import qualified Data.ByteArray as BA
+import qualified Data.List.NonEmpty as NE
+import qualified Network.Socket as Socket
+import qualified Ouroboros.Network.Block as Abstract
+import qualified Ouroboros.Network.Point as Point
+
+-- ----------------
+-- Development
+
+import Cardano.BM.Trace
+    ( nullTracer )
+import Control.Monad
+    ( (>=>) )
+
+convertTx :: CC.Tx -> W.Tx
+convertTx tx = W.Tx
+    { W.txId = convertHash $ CC.hash tx
+    , W.resolvedInputs = map convertTxIn $ NE.toList $ CC.txInputs tx
+    , W.outputs = map convertTxOut $ NE.toList $ CC.txOutputs tx
+    }
+  where
+    convertHash = W.Hash . BA.convert
+
+    convertTxIn (CC.TxInUtxo txH idx) = (W.TxIn (convertHash txH) idx, coin)
+       where
+          coin = error "1. We can't get the coin of Byron inputs!"
+
+    convertTxOut o = W.TxOut
+        (convertAddr $ CC.txOutAddress o)
+        (convertCoin $ CC.txOutValue o)
+      where
+        convertAddr = error "todo use toBinary"
+        -- Byron addresses should be no problem. I'm not sure how Shelley
+        -- addresses will look.
+        convertCoin = W.Coin . CC.unsafeGetLovelace
+
+convertBlockHeader :: CC.Block -> W.BlockHeader
+convertBlockHeader b = W.BlockHeader
+    { W.slotId =
+        error "2. can't easily get the SlotId"
+            -- The cardano-ledger equivalent is EpochAndSlotCount, i.e epoch
+            -- number and the slot counted from the start of the epoch.
+            --
+            -- But block headers only contain the slot number (SlotNo), counted
+            -- from genesis.
+    , W.blockHeight = error "3. We can't get the blockHeight"
```

Comment:
This is wrong. We [_can_](https://github.com/input-output-hk/ouroboros-network/blob/9d8493e6d609f879c9ebc684bc5d830f1f146bb6/ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron/Block.hs#L198) get it.